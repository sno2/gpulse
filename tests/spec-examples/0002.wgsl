diagnostic(off,derivative_uniformity);
var<private> d: f32;
fn helper() -> vec4<f32> {
  if (d < 0.5) {
    // The derivative_uniformity diagnostic is disabled here
    // by the global diagnostic filter.
    return textureSample(t,s,vec2(0,0));
  } else {
    // The derivative_uniformity diagnostic is set to 'warning' severity.
    @diagnostic(warning,derivative_uniformity) {
      return textureSample(t,s,vec2(0,0));
    }
  }
  return vec4(0.0);
}
