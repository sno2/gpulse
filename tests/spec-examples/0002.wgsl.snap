ast.Node{ .diagnostic_directive = ast.DiagnosticControl{ .severity = ast.SeverityControlName.off, .rule_namespace = ast.Span{ .start = 1, .end = 0 }, .rule_name = ast.Span{ .start = 15, .end = 36 } } }
ast.Node{ .var_decl = ast.VarDecl{ .name = ast.Span{ .start = 52, .end = 53 }, .access_mode = ast.AccessMode.read_write, .addr_space = ast.AddrSpace.private, .typ = ast.Node{ .identifier = ast.Span{ .start = 55, .end = 58 } }, .value = null } }
ast.Node{ .fn_decl = ast.FnDecl{ .name = ast.Span{ .start = 63, .end = 69 }, .params = {  }, .ret = ast.Node{ .template = ast.Template{ .name = ast.Span{ .start = 75, .end = 79 }, .args = { ast.Node{ .identifier = ast.Span{ .start = 80, .end = 83 } } } } }, .scope = { ast.Node{ .else_stmt = ast.CompoundStatement{ .attributes = null, .scope = { ast.Node{ .attributed = ast.Attributed{ .attributes = { ast.Attribute{ .diagnostic = ast.DiagnosticControl{ .severity = ast.SeverityControlName.warning, .rule_namespace = ast.Span{ .start = 1, .end = 0 }, .rule_name = ast.Span{ .start = 355, .end = 376 } } } }, .inner = ast.Node{ .scope = ast.Scope{ .scope = { ast.Node{ .ret = ast.Ret{ .value = ast.Node{ .call = ast.Call{ .callee = ast.Node{ .identifier = ast.Span{ .start = 393, .end = 406 } }, .args = { ast.Node{ .identifier = ast.Span{ .start = 407, .end = 408 } }, ast.Node{ .identifier = ast.Span{ .start = 409, .end = 410 } }, ast.Node{ .call = ast.Call{ .callee = ast.Node{ .identifier = ast.Span{ .start = 411, .end = 415 } }, .args = { ast.Node{ .number_literal = ast.NumberLiteral{ .span = ast.Span{ .start = 416, .end = 417 }, .kind = ast.NumberLiteral.NumberKind.abstract_int } }, ast.Node{ .number_literal = ast.NumberLiteral{ .span = ast.Span{ .start = 418, .end = 419 }, .kind = ast.NumberLiteral.NumberKind.abstract_int } } } } } } } } } } } } } } } } } }, ast.Node{ .ret = ast.Ret{ .value = ast.Node{ .call = ast.Call{ .callee = ast.Node{ .identifier = ast.Span{ .start = 442, .end = 446 } }, .args = { ast.Node{ .number_literal = ast.NumberLiteral{ .span = ast.Span{ .start = 447, .end = 450 }, .kind = ast.NumberLiteral.NumberKind.abstract_float } } } } } } } } } }
